pkg/machine/vmconfigs/machine.go

```diff
@@ -125,7 +118,7 @@ func (mc *MachineConfig) Refresh() error {
 }
 
 // write is a non-locking way to write the machine configuration file to disk
-func (mc *MachineConfig) write() error {
+func (mc *MachineConfig) Write() error {
```

Comment:
This change is problematic IMO, there are still many places in cmd/podman/machine that write a config. So now they write unlocked which can be a problem since multiple process could write at the same time causes incorrect status update.

I think we really have to follow a strict locking model. After we lock we should always re-read the config file, and only ever write/change the config while we hold the lock.

I think most of the unlock write can be solved by moving the write in the shim level were you now correctly lock.